sklearn/svm/classes.py

```diff
@@ -1128,3 +1130,169 @@ def predict(self, X):
+
+    Attributes
+    ----------
+    support_ : array-like, shape = [n_SV]
+        Indices of support vectors.
+
+    support_vectors_ : array-like, shape = [nSV, n_features]
+        Support vectors.
+
+    dual_coef_ : array, shape = [n_classes-1, n_SV]
+        Coefficients of the support vectors in the decision function.
+
+    coef_ : array, shape = [n_classes-1, n_features]
+        Weights assigned to the features (coefficients in the primal
+        problem). This is only available in the case of a linear kernel.
+
+        `coef_` is readonly property derived from `dual_coef_` and
+        `support_vectors_`
+
+    intercept_ : array, shape = [n_classes-1]
+        Constants in decision function.
+
+    References
+    ----------
+    .. [1] Tax, D.M. and Duin, R.P., 2004. "Support vector data
+           description." Machine learning, 54(1), pp.45-66.
+           doi:10.1023/B:MACH.0000008084.60811.49
+
+    .. [2] Chang, W.C., Lee, C.P. and Lin, C.J., 2013. "A revisit
+           to support vector data description (SVDD)." Technical
+           Report, Department of Computer Science, National Taiwan
+           University.
+    """
+    def __init__(self, kernel='rbf', degree=3, gamma='auto', coef0=0.0,
+                 tol=1e-3, nu=0.5, shrinking=True, cache_size=200,
+                 verbose=False, max_iter=-1, random_state=None):
+        super(SVDD, self).__init__(
+            'svdd_l1', kernel, degree, gamma, coef0, tol, 0., nu, 0.,
+            shrinking, False, cache_size, None, verbose, max_iter,
+            random_state)
+
+    def fit(self, X, y=None, sample_weight=None, **params):
+        """Detects the soft minimum volume hypersphere around the sample X.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape (n_samples, n_features)
+            Set of samples, where n_samples is the number of samples and
+            n_features is the number of features.
+
+        sample_weight : array-like, shape (n_samples,)
+            Per-sample weights. Higher weights force the novelty detector
+            to put more emphasis on these points.
+
+        Returns
+        -------
+        self : object
+            Returns self.
+
+        Notes
+        -----
+        If X is not a C-ordered contiguous array it is copied.
+
+        """
+        super(SVDD, self).fit(X, np.ones(_num_samples(X)),
+                              sample_weight=sample_weight, **params)
+        return self
+
+    def decision_function(self, X):
+        """Signed distance to the enveloping hypersphere.
+
+        Signed distance is positive for an inlier and negative for an outlier.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+
+        Returns
+        -------
+        X : array-like, shape (n_samples,)
+            Returns the decision function of the samples.
+        """
+        dec = self._decision_function(X)
+        return dec
+
```

Comment:
Can you add a `score_samples` method for consistency with other outlier detection estimators (PR #9015)
```python
    def score_samples(self, X):
         """Raw scoring function of the samples.
 
         Parameters
         ----------
         X : array-like, shape (n_samples, n_features)
 
         Returns
         -------
         score_samples : array-like, shape (n_samples,)
             Returns the (unshifted) scoring function of the samples.
         """
         score_samples = self.decision_function(X) + self.offset_
         return score_samples
```